src/fdosecrets/objects/Prompt.cpp

```diff
@@ -260,4 +333,87 @@ namespace FdoSecrets
         return {};
     }
 
+    OverwritePrompt::OverwritePrompt(Service* parent,
+                                     Connection* conn,
+                                     Item* item,
+                                     const QVariantMap& properties,
+                                     const SecretStruct& secret)
+        : PromptBase(parent)
+        , m_conn(conn)
+        , m_item(item)
+        , m_properties(properties)
+        , m_secret(secret)
+    {
+    }
+
+    DBusReturn<void> OverwritePrompt::prompt(const QString& windowId)
+    {
+        if (thread() != QThread::currentThread()) {
+            DBusReturn<void> ret;
+            QMetaObject::invokeMethod(this,
+                                      "prompt",
+                                      Qt::BlockingQueuedConnection,
+                                      Q_ARG(QString, windowId),
+                                      Q_RETURN_ARG(DBusReturn<void>, ret));
+            return ret;
+        }
+
+        MessageBox::OverrideParent override(findWindow(windowId));
+
+        bool dismissed;
+        if (m_conn && m_conn->connected() && m_item) {
+            auto l = m_item->locked(m_conn);
+            if (l.isError()) {
+                dismissed = true;
+            } else if (!l.value()) {
+                // not locked anymore
+                dismissed = false;
+            } else {
+                QString q = tr("An application wants to the update the following entry: %1\n\n")
+                                .arg(m_item->backend()->title());
+                q += tr("\nApplication: %1\n\n").arg(m_conn->peerName());
+                q += tr("Do you allow access to this entry only, all entries or deny the access?");
+
+                auto* checkbox = new QCheckBox(tr("Always request confirmations when applications access secrets"));
```

Comment:
` auto* checkbox = new QCheckBox(tr("Always request confirmations when applications access secrets"));`

What's the point of having this on the prompt? it's already on the settings, having it on the prompt is confusing